scipy/stats/_continuous_distns.py

```diff
@@ -3282,6 +3283,306 @@ def _ppf(self, q, k, s):
+                      1.0 - k*x)**(1.0/k))
+               logpdf = ...
+            '''
+            return special.xlog1py(1.0/k-1.0,-k*x)-(1.0-k*x)**(1.0/k)
+
+        def f2(x, h, k):
+            '''pdf = exp(-x)*(1.0 - h*exp(-x))**(1.0/h - 1.0)
+               logpdf = ...
+            '''
+            return -x + special.xlog1py(1.0/h-1.0, -h*exp(-x))
+
+        def f3(x, h, k):
+            '''pdf = exp(-x-exp(-x))
+               logpdf = ...
+            '''
+            return -x-exp(-x)
+
+        return _lazyselect(condlist,
+                           [f0, f1, f2, f3],
+                           [x, h, k],
+                           default=nan)
+
+    def _cdf(self, x, h, k):
+        return exp(self._logcdf(x, h, k))
+
+    def _logcdf(self, x, h, k):
+        condlist = [np.logical_and(h != 0, k != 0),
+                    np.logical_and(h == 0, k != 0),
+                    np.logical_and(h != 0, k == 0),
+                    np.logical_and(h == 0, k == 0)]
+
+        def f0(x, h, k):
+            '''cdf = (1.0 - h*(1.0 - k*x)**(1.0/k))**(1.0/h)
+               logcdf = ...
+            '''
+            return (1.0/h)*special.log1p(-h*(1.0-k*x)**(1.0/k))
+
+        def f1(x, h, k):
+            '''cdf = exp(-(1.0 - k*x)**(1.0/k))
+               logcdf = ...
+            '''
+            return -(1.0 - k*x)**(1.0/k)
+
+        def f2(x, h, k):
+            '''cdf = (1.0 - h*exp(-x))**(1.0/h)
+               logcdf = ...
+            '''
+            return (1.0/h)*special.log1p(-h*exp(-x))
+
+        def f3(x, h, k):
+            '''cdf = exp(-exp(-x))
+               logcdf = ...
+            '''
+            return -exp(-x)
+
+        return _lazyselect(condlist,
+                           [f0, f1, f2, f3],
+                           [x, h, k],
+                           default=nan)
+
+    def _ppf(self, q, h, k):
+        condlist = [np.logical_and(h != 0, k != 0),
+                    np.logical_and(h == 0, k != 0),
+                    np.logical_and(h != 0, k == 0),
+                    np.logical_and(h == 0, k == 0)]
+
+        def f0(q, h, k):
+            return 1.0/k*(1.0 - ((1.0 - (q**h))/h)**k)
+
+        def f1(q, h, k):
+            return 1.0/k*(1.0 - (-log(q))**k)
+
+        def f2(q, h, k):
+            '''ppf = -log((1.0 - (q**h))/h)
+            '''
+            return -special.log1p(-(q**h)) + log(h)
+
+        def f3(q, h, k):
+            return -log(-log(q))
+
+        return _lazyselect(condlist,
+                           [f0, f1, f2, f3],
+                           [q, h, k],
+                           default=nan)
+
+    def _stats(self, h, k):
+        if h == 0 or k == 0:
+            maxr = 0
+        elif h < 0 and k > 0:
+            maxr = int(-1.0/h*k)
+        elif k < 0:
+            maxr = int(-1.0/k)
+        else:
+            maxr = 5
+
+        outputs = [None if r < maxr else nan for r in range(1, 5)]
+        return outputs[:]
```

Comment:
This looks strange: `None` signals "moment exists, defer to a generic calculation". 
For `h=k=0`, the moments do exist, but this function returns nans. Not sure about other cases.

```
In [14]: stats.kappa4.stats(h, k, moments='m')
Out[14]: array(nan)

In [15]: stats.kappa4.expect(lambda x: x, args=(h, k))
Out[15]: 0.577215664901533

In [16]: h, k
Out[16]: (0, 0)
```